test/e2e/mcd_test.go

```diff
@@ -322,6 +327,35 @@ func TestKernelType(t *testing.T) {
 		}
 		t.Logf("Node %s has expected kernel", node.Name)
 	}
+
+	// Delete the applied kerneltype MachineConfig to make sure rollback works fine
+	if err := cs.MachineConfigs().Delete(kernelType.Name, &metav1.DeleteOptions{}); err != nil {
+		t.Error(err)
+	}
+
+	t.Logf("Deleted MachineConfig %s", kernelType.Name)
+
+	// Wait for the mcp to rollback to previous config
+	if err := waitForPoolComplete(t, cs, "worker", workerOldMC); err != nil {
+		t.Fatal(err)
+	}
+	for _, node := range nodes {
```

Comment:
oh yeah, this is what's happening - you're using the `nodes` slice from a previous fetch, you need to add:

```go
nodes, err := getNodesByRole(cs, "worker")
require.Nil(t, err)
```

So it grabs a fresh slice with the correct annotations

Reply:
ah, I didn't fetch it again because I thought we should be getting same worker nodes slice during re-fetch. But yeah,  annotations would have changed.